genkan/auth/oAuth.js

```diff
@@ -0,0 +1,182 @@
+const saltRounds = 10;
+
+// Random Password Generator
+const generator = require('generate-password');
+
+// Token Generator
+const tokenGenerator = require('./tokenGenerator')
+
+// NOTE: Will clean up and organise the code here
+
+MongoClient.connect(url, {useUnifiedTopology: true}, function(err, client) {
+    if (err) throw err
+
+    const db = client.db(dbName)
+
+    loginAccountGoogle = (email, googleID, callback) => {
+    // Checks if user has google id in db
+        findDB(db, 'users', {'email': email, 'googleID': googleID}, (result) => {
+            // If exists
+            // console.log(result)
+            if (result.length !== 0) {
+                // Generate a random token for SID
+                const sid = tokenGenerator()
+
+                // Schema for sessions in session collection
+                const SessionSchema = {
+                    'uid': result[0]._id,
+                    'sid': tokenGenerator(),
+                    'timestamp': (new Date()).toISOString(),
+                    'createdTimestamp': new Date(),
+                }
+                // Payload to update user's last seen in users collection
+                const UpdateLastSeenPayload = {
+                    $set: {
+                        'account.activity.lastSeen': new Date(),
+                    },
+                }
+
+                // Update database
+                insertDB(db, 'sessions', SessionSchema, () => {
+                    updateDB(db, 'users', {'email': email}, UpdateLastSeenPayload, () => {
+                        return callback(sid)
+                    })
+                })
+
+                // If google ID or email of user is not in db
+                // That means user is new or probably has existing email but no google ID present
+            } else {
+                // If user has email in db but google ID is not present
+                // We update db to include googleID
+                findDB(db, 'users', {'email': email}, (result) => {
+                    // If user has email in db
+
+                    if (result.length !== 0) {
+                        // Generate a random token for SID
+                        const sid = tokenGenerator()
+
+                        // Schema for sessions in session collection
+                        const SessionSchema = {
+                            'uid': result[0]._id,
+                            'sid': tokenGenerator(),
+                            'timestamp': (new Date()).toISOString(),
+                            'createdTimestamp': new Date(),
+                        }
+                        // Payload to update user's last seen in users collection
+                        const UpdateLastSeenPayload = {
+                            $set: {
+                                'account.activity.lastSeen': new Date(),
+                            },
+                        }
+
+                        const setGoogleID = {
```

Comment:
Would be better to start payload and schema names with a capital letter to differ them from usual objects.
--> `SetGoogleID`